app/services/slack_service.rb

```diff
@@ -0,0 +1,183 @@
+module SlackService
```

Comment:
Is there any reason you didn't use the Ruby Slack client? https://github.com/slack-ruby/slack-ruby-client

This class is doing a lot of complicated low level text processing which I think the Slack libraries could handle for you? Moreover Slack even has a whole framework for writing these kinds of bots.

Reply:
I thought the logic was simple enough to not use a framework but if you disagree we could use the gem you mentioned above.

Reply:
I think it's always good to think twice about using a library.

Using a library can certainly have drawbacks, as a library usually has a lot more code than what you need so a bigger surface for bugs and security vulnerabilities, a library can also hold you back when upgrade to a different Ruby version etc.

In this case though I would say using a Slack library makes sense. The amount of error prone, quite low level, code you're writing definitely justifies it. The library also seems to be well supported.